kani-compiler/src/kani_middle/attributes.rs

```diff
@@ -228,6 +373,74 @@ impl<'tcx> KaniAttributes<'tcx> {
     }
 }
 
+/// A basic check that ensures a function with a contract does not receive
+/// mutable pointers in its input and does not return raw pointers of any kind.
+///
+/// This is a temporary safety measure because contracts cannot yet reason
+/// about the heap.
+fn check_is_contract_safe(tcx: TyCtxt, item: DefId) {
```

Comment:
How do you know if a generic parameter is not specialized into a pointer?

Reply:
What if we move this function to either `contracts.rs` or to `kani_middle/mod.rs` and change the input to be a `MonoItem`? Let's say something like:
```rust
fn check_is_contract_safe(tcx: TyCtxt, item: MonoItem)
```

you could then invoke this function from `check_reachable_item` [here](https://github.com/model-checking/kani/blob/596bc5085ac036f02459dd77823f8b6df27fbea1/kani-compiler/src/kani_middle/mod.rs#L68C11-L68C11).